drivers/Watchdog.cpp

```diff
@@ -20,6 +20,72 @@
 
 namespace mbed {
 
+uint32_t Watchdog::_is_initialized = 0;
+uint32_t Watchdog::_re_initialize = 0;
+uint32_t Watchdog::_bitmask = 0;
+uint32_t Watchdog::_kick_bitmask = 0;
+Watchdog::watchdog_usr_info Watchdog::_usr_info[MAX_THREAD_WATCHDOG_SUPPORT] = {0};
+rtos::Mutex Watchdog::_ThreadSafeLockMutex;
+mbed_error_status_t Watchdog::wd_register(const osThreadId_t tid, const uint32_t timeout)
+{
+    int store_index = -1;
+    watchdog_status_t sts;
+    int i;
+    _ThreadSafeLockMutex.lock();
+    uint32_t conf_timeout = hal_watchdog_get_reload_value();
+    if(_is_initialized) {
+        if(timeout > conf_timeout) {
+            _re_initialize = 1;
+            hal_watchdog_stop();
+         }
+
+        for(i =0 ; i < MAX_THREAD_WATCHDOG_SUPPORT; i++) {
+            if(_bitmask & (1 << i))
+            {
+                if(_usr_info[i].tid == tid) {
+                    if(_re_initialize) {
```

Comment:
Don't really understand this `_re_initialize` flag? Does this mean we fail if `timeout` < `conf_timeout`?